src/main/java/duke/command/DeleteTaskCommand.java

```diff
@@ -14,23 +14,23 @@
 
 public class DeleteTaskCommand extends Command{
     private int id;
-    private String command;
+    private String deletedTaskInfo;
 
-    public DeleteTaskCommand(String command) {
-        this.command = command;
+    public DeleteTaskCommand(String deletedTaskInfo) throws DukeException {
+
+        this.deletedTaskInfo = deletedTaskInfo;
+        if (Character.isDigit(deletedTaskInfo.charAt(0))) {
+            try {
+                this.id = Integer.parseInt(deletedTaskInfo);
+            } catch (Exception e) {
+                throw new DukeException("Please follow format 'delete task #<id>'. ");
```

Comment:
I think there is another problem with input command

-  `delete task 123`

The taskInfo `123` here will be identified as taskID instead of description. According to the format `delete task <taskInfo>`, it might lead to unexpected outcome.

Reply:
I suggest instead of doing
 `if (Character.isDigit(deletedTaskInfo.charAt(0))`,
we can create two constructors
1. `DeleteCommand (int info)`
2. `DeleteCommand (String info)`
Such that when `#` is replaced in Parser, we pass in an integer as argument

Reply:
Alright, I'll see if I can find an alternate method of checking for the id--the difficulty comes from preventing any further parsing from happening in the Command class (we don't want to have to check for # in a string, only to parse its string again), but I'm sure there's some sort of solution.

Thanks for the feedback!

Reply:
Its rather difficult to have an alternate solution for now (As Parser only returns a String).
Will discuss with you if I have any clue (Most likely not 😢)